AI/BattleAI/EnemyInfo.h

```diff
@@ -7,10 +7,11 @@
  * Full text of license available in license.txt file, in main folder
  *
  */
-
 #pragma once
-
-#include "ThreatMap.h"
+#include "common.h"
+#include "CCallback.h"
+#include "../../lib/BattleState.h"
+class CStack;
```

Comment:
Do not include BattleState.h and decalre CStack at the same time. Prefere to use forward declaration. and use inlude only if it is absolutely necessary. 
Actually you need _only_ include BattleHex.h here.